### PR TITLE
[CLEANUP] removed `providing reversed arguments to observer`

### PR DESCRIPTION
**Pull request by bekzod**

### CODE REVIEW

File: packages/ember-metal/lib/mixin.js [hunk 1]
```diff
@@ -842,22 +833,12 @@ export function _immediateObserver() {
   @private
 */
 export function _beforeObserver(...args) {
```
Comment (bekzod) [inline packages/ember-metal/lib/mixin.js]: removed providing reversed arguments to _beforeObserver too

**Comment by rwjblue** [comment]
Can you create a corresponding PR (or issue that links back here) at https://github.com/emberjs/ember-2-legacy? All of the API functionality we remove from this repo for 3.0 must be "re-enableable" by including that addon.

**Comment by bekzod** [comment]
done

**Comment by rwjblue** [comment]
Thank you!

**Pull request merged**